client/cody/src/chat/protocol.ts

```diff
@@ -57,12 +56,42 @@ export interface AuthStatus {
     authenticated: boolean
     hasVerifiedEmail: boolean
     requiresVerifiedEmail: boolean
+    onSupportedSiteVersion: boolean
 }
 
 export function isLoggedIn(authStatus: AuthStatus): boolean {
+    if (!authStatus.onSupportedSiteVersion) {
+        return false
+    }
     return authStatus.authenticated && (authStatus.requiresVerifiedEmail ? authStatus.hasVerifiedEmail : true)
 }
 
 export function isLocalApp(url: string): boolean {
     return new URL(url).origin === LOCAL_APP_URL.origin
 }
+
+// if version is below 5.1.0, it is not supported
+// the version number will be in this format "222587_2023-05-30_5.0-39cbcf1a50f0" for insider builds
+export function isSiteVersionSupported(version: string | Error): boolean {
+    const MAJOR_VERSION = 5
+    const MINOR_VERSION = 1
+    if (isError(version) || !version) {
+        return false
+    }
+    if (version.length > 10) {
```

Comment:
Maybe add an explaining variable name, because `> 10` is a bit mysterious and we're a few lines away from the context of the comment.

```suggestion
    const isInsiderBuild = version.length > 10
    if (isInsiderBuild) {
```